File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return ""; // Edge case: empty input
        
        sort(strs.begin(), strs.end()); // Sort lexicographically
        string first = strs[0];
        string last = strs[strs.size() - 1];
        
        string result = "";
        for (int i = 0; i < min(first.size(), last.size()); i++) {
            if (first[i] != last[i]) break;
            result += first[i];
        }
        return result;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String first = strs[0];
        String last = strs[strs.length-1];
        int i = 0;
        int j = 0;
        StringBuilder ans = new StringBuilder(); 
        while(i<first.length() && j<last.length()){
            if(first.charAt(i) != last.charAt(j)){
                break;
            }
            else if(first.charAt(i) == last.charAt(j)){
                ans.append(first.charAt(i));
                i++;
                j++;
            }
        }
        return ans.toString();
    }
}